src/main/java/org/takes/facets/auth/codecs/CcAES.java

```diff
@@ -145,6 +121,32 @@ public Identity decode(final byte[] bytes) throws IOException {
     }
 
     /**
+     * Create AlgorithmParameterSpec with the block size.
+     *
+     * @return The AlgorithmParameterSpec
+     */
+    private static AlgorithmParameterSpec initAlgorithmParameterSpec() {
+        final SecureRandom random = new SecureRandom();
+        final byte[] bytes = new byte[CcAES.BLOCK];
+        random.nextBytes(bytes);
+        return new IvParameterSpec(bytes);
+    }
+
+    /**
+     * Check the block size of the key.
+     */
+    private void checkBlockSize() {
```

Comment:
@sebing it's a bad practice, see http://www.yegor256.com/2015/12/08/temporal-coupling-between-method-calls.html